helper/predeployment/predeployment.go

```diff
@@ -0,0 +1,254 @@
+	}, nil
+}
+
+// getArraySubstring fetches the substring of anything
+// that is between [ and ]
+func getArraySubstring(s string) string {
+	// Find the first occurrence of the opening bracket
+	i := strings.Index(s, "[")
+	if i >= 0 {
+		// Find last occurrence of the closing bracket
+		j := strings.LastIndex(s, "]")
+		if j >= 0 {
+			return s[i+1 : j]
+		}
+	}
+
+	return ""
+}
+
+// extractValue recursively extracts the values for subarrays
+// and packs them
+func extractValue(s string) interface{} {
+	if !strings.Contains(s, "[") {
+		return s
+	}
+
+	ret := make([]interface{}, 0)
+	arraySubstring := getArraySubstring(s)
+
+	if arraySubstring != "" {
+		splitValues := strings.Split(arraySubstring, ",")
+
+		for _, str := range splitValues {
+			ret = append(ret, extractValue(str))
+		}
+	}
+
+	return ret
+}
+
+// normalizeConstructorArguments cleans up the constructor arguments so that they
+// can be interpreted by the ABI encoder
+func normalizeConstructorArguments(constructorArgs []interface{}) []interface{} {
+	arguments := make([]interface{}, 0)
+
+	for _, arg := range constructorArgs {
+		argStr, _ := arg.(string)
+		arguments = append(arguments, extractValue(argStr))
+	}
+
+	return arguments
+}
+
+// GenerateGenesisAccountFromFile generates an account that is going to be directly
+// inserted into state
+func GenerateGenesisAccountFromFile(
+	filepath string,
+	constructorArgs []interface{},
+) (*chain.GenesisAccount, error) {
+	// Create the artifact from JSON
+	artifact, err := generateContractArtifact(filepath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Generate the contract ABI object
+	contractABI, err := abi.NewABI(string(artifact.ABI))
+	if err != nil {
+		return nil, fmt.Errorf("unable to create contract ABI, %w", err)
+	}
+
+	// Encode the constructor params
+	constructor, err := abi.Encode(
+		// Constructor arguments are passed in as an array of values.
+		// Structs are treated as sub-arrays with their corresponding values laid out
+		// in ABI encoding
+		normalizeConstructorArguments(constructorArgs),
+		contractABI.Constructor.Inputs,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("unable to encode constructor arguments, %w", err)
+	}
+
+	finalBytecode := append(artifact.Bytecode, constructor...)
+
+	// Create an instance of the state
+	st := itrie.NewState(itrie.NewMemoryStorage())
+
+	// Create a snapshot
+	snapshot := st.NewSnapshot()
+
+	// Create a radix
+	radix := state.NewTxn(st, snapshot)
+
+	// Create the contract object for the EVM
+	contract := runtime.NewContractCreation(
+		1,
+		types.ZeroAddress,
+		types.ZeroAddress,
+		staking.AddrStakingContract,
+		big.NewInt(0),
+		math.MaxInt64,
+		finalBytecode,
+	)
+
+	// Enable all forks
+	config := chain.ForksInTime{
+		Homestead:      true,
+		Byzantium:      true,
+		Constantinople: true,
+		Petersburg:     true,
+		Istanbul:       true,
+		EIP150:         true,
+		EIP158:         true,
+		EIP155:         true,
+	}
```

Comment:
Shall we define this as a constant varriable?

Reply:
Perhaps it will be same as `chain.AllForksEnabled.At(0)`

Reply:
This is a really good observation 💯 

Added:

[3acfcbf](https://github.com/0xPolygon/polygon-edge/pull/536/commits/3acfcbf848d69b09d725e044e394bd2e9f2bca7b)